src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/state/Task.kt

```diff
@@ -0,0 +1,184 @@
+      else -> null
+    }
+  },
+
+  COMPLETE_2_DAY_CHECKS {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, ELIGIBLE_AND_SUITABLE, AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE, OPTED_OUT -> LOCKED
+      else -> null
+    }
+  },
+
+  PRINT_LICENCE {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, ELIGIBLE_AND_SUITABLE, AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE, OPTED_OUT -> LOCKED
+      else -> null
+    }
+  },
+
+  CONSULT_THE_VLO_AND_POM {
+    override fun visibleForRoles() = setOf(PROBATION_COM)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      AWAITING_ADDRESS_AND_RISK_CHECKS ->
+        if (assessment.victimContactSchemeOptedIn != null) COMPLETE else READY_TO_START
+
+      ADDRESS_AND_RISK_CHECKS_IN_PROGRESS -> COMPLETE
+      else -> null
+    }
+  },
+
+  CHECK_ADDRESSES_OR_COMMUNITY_ACCOMMODATION {
+    override fun visibleForRoles() = setOf(PROBATION_COM)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      AWAITING_ADDRESS_AND_RISK_CHECKS -> READY_TO_START
+      ADDRESS_AND_RISK_CHECKS_IN_PROGRESS -> IN_PROGRESS
+      else -> null
+    }
+  },
+
+  RECORD_NON_DISCLOSABLE_INFORMATION {
+    override fun visibleForRoles() = setOf(PROBATION_COM)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS ->
+        if (assessment.hasNonDisclosableInformation != null) COMPLETE else READY_TO_START
+
+      else -> null
+    }
+  },
+
+  MAKE_A_RISK_MANAGEMENT_DECISION {
+    override fun visibleForRoles() = setOf(PROBATION_COM)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS -> LOCKED
+      else -> null
+    }
+  },
+
+  SEND_CHECKS_TO_PRISON {
+    override fun visibleForRoles() = setOf(PROBATION_COM)
+    override fun taskStatus(assessment: Assessment): TaskStatus? {
+      if (!assessment.status.isOneOf(AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS)) {
+        return null
+      }
+      return when {
+        assessment.addressChecksStatus == ResidentialChecksStatus.UNSUITABLE -> READY_TO_START
+        assessment.addressChecksStatus == ResidentialChecksStatus.SUITABLE && assessment.victimContactSchemeOptedIn != null && assessment.hasNonDisclosableInformation != null -> READY_TO_START
```

Comment:
previously this logic had to be shared across multiple states, now it only needs to be specified once